exporter/awsemfexporter/grouped_metric.go

```diff
@@ -39,68 +38,70 @@ type metricInfo struct {
 
 // addToGroupedMetric processes OT metrics and adds them into GroupedMetric buckets
 func addToGroupedMetric(pmd pmetric.Metric, groupedMetrics map[interface{}]*groupedMetric, metadata cWMetricMetadata, patternReplaceSucceeded bool, logger *zap.Logger, descriptor map[string]MetricDescriptor, config *Config) error {
-	metricName := pmd.Name()
+
 	dps := getDataPoints(pmd, metadata, logger)
 	if dps == nil || dps.Len() == 0 {
 		return nil
 	}
 
 	for i := 0; i < dps.Len(); i++ {
-		dp, retained := dps.At(i)
+		dps, retained := dps.CalculateDeltaDatapoints(i, metadata.instrumentationLibraryName, config.DetailedMetrics)
 		if !retained {
 			continue
 		}
 
-		labels := dp.labels
+		for _, dp := range dps {
+			labels := dp.labels
 
-		if metricType, ok := labels["Type"]; ok {
-			if (metricType == "Pod" || metricType == "Container") && config.EKSFargateContainerInsightsEnabled {
-				addKubernetesWrapper(labels)
+			if metricType, ok := labels["Type"]; ok {
+				if (metricType == "Pod" || metricType == "Container") && config.EKSFargateContainerInsightsEnabled {
+					addKubernetesWrapper(labels)
+				}
 			}
-		}
 
-		// if patterns were found in config file and weren't replaced by resource attributes, replace those patterns with metric labels.
-		// if patterns are provided for a valid key and that key doesn't exist in the resource attributes, it is replaced with `undefined`.
-		if !patternReplaceSucceeded {
-			if strings.Contains(metadata.logGroup, "undefined") {
-				metadata.logGroup, _ = replacePatterns(config.LogGroupName, labels, logger)
-			}
-			if strings.Contains(metadata.logStream, "undefined") {
-				metadata.logStream, _ = replacePatterns(config.LogStreamName, labels, logger)
+			// if patterns were found in config file and weren't replaced by resource attributes, replace those patterns with metric labels.
+			// if patterns are provided for a valid key and that key doesn't exist in the resource attributes, it is replaced with `undefined`.
+			if !patternReplaceSucceeded {
+				if strings.Contains(metadata.logGroup, "undefined") {
+					metadata.logGroup, _ = replacePatterns(config.LogGroupName, labels, config.logger)
+				}
+				if strings.Contains(metadata.logStream, "undefined") {
+					metadata.logStream, _ = replacePatterns(config.LogStreamName, labels, config.logger)
+				}
 			}
-		}
 
-		metric := &metricInfo{
-			value: dp.value,
-			unit:  translateUnit(pmd, descriptor),
-		}
+			metric := &metricInfo{
+				value: dp.value,
+				unit:  translateUnit(pmd, descriptor),
+			}
```

Comment:
Are these spaces and not tabs?  Be sure to run `go fmt`.

Reply:
Yes I had already run `go fmt`.